onnxruntime/core/providers/cpu/tensor/slice.cc

```diff
@@ -8,27 +8,64 @@ using namespace std;
 
 namespace onnxruntime {
 
-#define ADD_TYPED_SLICE_OP(data_type)                                                   \
+#define ADD_TYPED_SLICE_OP(data_type, indice_type)                                      \
   ONNX_CPU_OPERATOR_TYPED_KERNEL(                                                       \
       Slice,                                                                            \
       1,                                                                                \
       data_type,                                                                        \
       KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<data_type>()), \
-      Slice<data_type>);
-
-ADD_TYPED_SLICE_OP(uint8_t);
-ADD_TYPED_SLICE_OP(uint16_t);
-ADD_TYPED_SLICE_OP(uint32_t);
-ADD_TYPED_SLICE_OP(uint64_t);
-ADD_TYPED_SLICE_OP(int8_t);
-ADD_TYPED_SLICE_OP(int16_t);
-ADD_TYPED_SLICE_OP(int32_t);
-ADD_TYPED_SLICE_OP(int64_t);
-ADD_TYPED_SLICE_OP(float);
-ADD_TYPED_SLICE_OP(double);
-ADD_TYPED_SLICE_OP(MLFloat16);
-ADD_TYPED_SLICE_OP(bool);
-ADD_TYPED_SLICE_OP(string);
+      Slice<data_type, indice_type, false>);
+
+ADD_TYPED_SLICE_OP(uint8_t,  int64_t);
+ADD_TYPED_SLICE_OP(uint16_t, int64_t);
+ADD_TYPED_SLICE_OP(uint32_t, int64_t);
+ADD_TYPED_SLICE_OP(uint64_t, int64_t);
+ADD_TYPED_SLICE_OP(int8_t,   int64_t);
+ADD_TYPED_SLICE_OP(int16_t,  int64_t);
+ADD_TYPED_SLICE_OP(int32_t,  int64_t);
+ADD_TYPED_SLICE_OP(int64_t,  int64_t);
+ADD_TYPED_SLICE_OP(float,    int64_t);
+ADD_TYPED_SLICE_OP(double,   int64_t);
+ADD_TYPED_SLICE_OP(MLFloat16,int64_t);
+ADD_TYPED_SLICE_OP(bool,     int64_t);
+ADD_TYPED_SLICE_OP(string,   int64_t);
+
+#define ADD_TYPED_DYNAMIC_SLICE_OP(data_type, indice_type)                                   \
+  ONNX_CPU_OPERATOR_TYPED_KERNEL(                                                            \
+      DynamicSlice,                                                                          \
+      1,                                                                                     \
+      data_type##_##indice_type,                                                             \
+      KernelDefBuilder().TypeConstraint("T",    DataTypeImpl::GetTensorType<data_type>())    \
+                        .TypeConstraint("Tind", DataTypeImpl::GetTensorType<indice_type>()), \
+      Slice<data_type, indice_type>);
+
+ADD_TYPED_DYNAMIC_SLICE_OP(uint8_t,  int32_t);
```

Comment:
Did you implement indice_type as int32_t ? I guess no.
Then:
1. don't register for these types
2. why the build didn't fail?